src/proc/y411-converter.cpp

```diff
@@ -0,0 +1,311 @@
+            __m128i rgba_24_32_l0 = _mm_unpacklo_epi64(rgba_24_27, rgba_28_32);
+            __m128i rgba_24_32_l1 = _mm_unpackhi_epi64(rgba_24_27, rgba_28_32);
+
+            // Shuffle rgb triples to the start and end of each register
+            __m128i rgb0_l0 = _mm_shuffle_epi8(rgba_0_7_l0, _mm_setr_epi8(3, 7, 11, 15, 0, 1, 2, 4, 5, 6, 8, 9, 10, 12, 13, 14));
+            __m128i rgb1_l0 = _mm_shuffle_epi8(rgba_8_15_l0, _mm_setr_epi8(0, 1, 2, 4, 3, 7, 11, 15, 5, 6, 8, 9, 10, 12, 13, 14));
+            __m128i rgb2_l0 = _mm_shuffle_epi8(rgba_16_23_l0, _mm_setr_epi8(0, 1, 2, 4, 5, 6, 8, 9, 3, 7, 11, 15, 10, 12, 13, 14));
+            __m128i rgb3_l0 = _mm_shuffle_epi8(rgba_24_32_l0, _mm_setr_epi8(0, 1, 2, 4, 5, 6, 8, 9, 10, 12, 13, 14, 3, 7, 11, 15));
+
+
+            // calculate the current line and column
+            auto num_on_regs_at_once = 3;
+            auto rgb_bpp = 3;
+            auto reg_num_on_line = w * rgb_bpp / 16;
+            auto line = (i*num_on_regs_at_once) / reg_num_on_line;
+            auto j = i % (reg_num_on_line / num_on_regs_at_once);
+
+            // Align registers and store 16 pixels (48 bytes) at once on the line above
+            _mm_storeu_si128(&dst[(line*2 ) *reg_num_on_line + j * 3], _mm_alignr_epi8(rgb1_l0, rgb0_l0, 4));
+            _mm_storeu_si128(&dst[(line*2 ) * reg_num_on_line + j * 3 + 1], _mm_alignr_epi8(rgb2_l0, rgb1_l0, 8));
+            _mm_storeu_si128(&dst[(line*2 ) * reg_num_on_line + j * 3 + 2], _mm_alignr_epi8(rgb3_l0, rgb2_l0, 12));
+
+            // Shuffle rgb triples to the start and end of each register
+            __m128i rgb0_l1 = _mm_shuffle_epi8(rgba_0_7_l1, _mm_setr_epi8(3, 7, 11, 15, 0, 1, 2, 4, 5, 6, 8, 9, 10, 12, 13, 14));
+            __m128i rgb1_l1 = _mm_shuffle_epi8(rgba_8_15_l1, _mm_setr_epi8(0, 1, 2, 4, 3, 7, 11, 15, 5, 6, 8, 9, 10, 12, 13, 14));
+            __m128i rgb2_l1 = _mm_shuffle_epi8(rgba_16_23_l1, _mm_setr_epi8(0, 1, 2, 4, 5, 6, 8, 9, 3, 7, 11, 15, 10, 12, 13, 14));
+            __m128i rgb3_l1 = _mm_shuffle_epi8(rgba_24_32_l1, _mm_setr_epi8(0, 1, 2, 4, 5, 6, 8, 9, 10, 12, 13, 14, 3, 7, 11, 15));
+
+            // Align registers and store 16 pixels(48 bytes) at once on the line bellow
+            _mm_storeu_si128(&dst[(line*2  + 1) *reg_num_on_line + j * 3], _mm_alignr_epi8(rgb1_l1, rgb0_l1, 4));
+            _mm_storeu_si128(&dst[(line*2  + 1) * reg_num_on_line + j * 3 + 1], _mm_alignr_epi8(rgb2_l1, rgb1_l1, 8));
+            _mm_storeu_si128(&dst[(line*2  + 1) * reg_num_on_line + j * 3 + 2], _mm_alignr_epi8(rgb3_l1, rgb2_l1, 12));
+        }
+#endif
+    }
+    void unpack_y411_native(byte * const dest, const byte * s, int w, int h, int actual_size)
+    {
+        auto out = dest;
+
+        auto index_source = 0;
+        for (auto i = 0; i < h; i += 2)
+        {
+            for (auto j = 0; j < w; j += 2)
+            {
+                auto y411_pix = &s[index_source];
+                auto l0_u0 = y411_pix[0];
```

Comment:
Remove `l0_` for `u` and `v` (no `u0` or `v0`)

Reply:
done